docs: user onboarding, starting a new workspace, URL parameters (#2252)
codenvy/che-docs: modules/end-user-guide/pages/basic-actions-you-can-perform-on-a-workspace.adoc

:_content-type: REFERENCE
:description: Basic actions you can perform on a workspace

> :keywords: running-workspace, stopped-workspace, open-workspace, open-a-workspace, open-a-running-workspace, opening-workspace, opening-a-workspace, opening-a-running-workspace, how-to-open-workspace, how-to-open-a-workspace, how-to-open-a-running-workspace, reopen-workspace, reopen-a-workspace, reopen-a-running-workspace, reopening-workspace, reopening-a-workspace, reopening-a-running-workspace, how-to-reopen-workspace, how-to-reopen-a-workspace, how-to-reopen-a-running-workspace, restart-workspace, restart-a-workspace, restart-a-running-workspace, restarting-workspace, restarting-a-workspace, restarting-a-running-workspace, how-to-restart-workspace, how-to-restart-a-workspace, how-to-restart-a-running-workspace, stop-workspace, stop-a-workspace, stop-a-running-workspace,  stopping-workspace, stopping-a-workspace, stopping-a-running-workspace, how-to-stop-workspace, how-to-stop-a-workspace, how-to-stop-a-running-workspace, start-workspace, start-a-workspace, start-stopped-workspace, start-a-stopped-workspace, starting-workspace, starting-a-workspace, starting-stopped-workspace, starting-a-stopped-workspace, how-to-start-workspace, how-to-start-a-workspace, how-to-start-stopped-workspace, how-to-start-a-stopped-workspace, delete-workspace, delete-a-workspace, deleting-workspace, deleting-a-workspace, how-to-delete-workspace, how-to-delete-a-workspace

Reviewer: "Keyword stuffing" might decrease the SEO quality of your content: https://developers.google.com/search/docs/advanced/guidelines/irrelevant-keywords

Author: Thanks for the heads up. In this case, all the keywords are descriptive of the actual content of the page. I ended up with such a long list because I previously had the subsections here as separate pages, where each page had its own list. I ended up "merging" the several pages into one page, and I just added up the keyboards.

The quantity of keyboards aside, my main motivator for adding keywords is making sure that the combinations that readers may type in the Search field in our docs pages pick up the relevant page.

Reviewer 2: This list of keywords is not necessary at all. The search engine is perfectly capable of resolving queries with and without articles, using the gerund or infinitive forms, etc. Therefore, including, for example, the following is not helping anything, and Rolfe's comment applies:

```
restart-workspace, restart-a-workspace, restart-a-running-workspace, restarting-workspace, restarting-a-workspace, restarting-a-running-workspace, how-to-restart-workspace, how-to-restart-a-workspace, how-to-restart-a-running-workspace
```

The same goes for stopping, starting, opening, etc. Remove this, please, and keep the keyword list to four or five items.

Author: Fixed. Thank you both.

:navtitle: Basic actions you can perform on a workspace
// :page-aliases:

[id="basic-actions-you-can-perform-on-a-workspace_{context}"]
= Basic actions you can perform on a workspace

> You manage your workspaces and verify their current states in the *Workspaces* page (`{prod-url}/dashboard/#/workspaces`) of your {prod-short} dashboard.
> 
> After you start a new workspace, you can perform the following actions on it in the *Workspaces* page:

Reviewer: ```suggestion
After you start a new workspace, you can stop, start, reopen, restart, and delete it on the *Workspaces* page (`{prod-url}/dashboard/#/workspaces`) of your {prod-short} dashboard.
```

Reviewer: In the openshift-docs repo (not the eclipse-che docs) I see references to `*Workspaces* section` instead of `page`. It might be good to review this and "harmonize" our nomenclature, eventually.

Author: Let's look at this: by *Workspaces* page in this sentence I mean the web application GUI of CodeReady Workspaces.

I've jus tried searching for *Workspaces*:
https://github.com/openshift/openshift-docs/search?q=*Workspaces*
And I assume this is what you meant:
https://github.com/openshift/openshift-docs/blob/0fd09865e36c7eb97a8230a661632f62866a30c5/modules/op-constructing-pipelines-using-pipeline-builder.adoc
And this result appears to be describing YAML content sections.
If so, then we needn't worry about harmonizing.

BTW, I've discovered some outdated content in the search results and already opened a Jira issue in our docs for our team to get in touch with the OpenShift writers. This is outdated: https://github.com/openshift/openshift-docs/blob/b0199378013f0684a693f66c5bdb5dd430642c41/modules/codeready-workspaces.adoc

Author: @rolfedh, i appreciate your suggestion. The problem for me here and the reason why I added the table is the fact that the current GUI implementation unfortunately needs explaining. The second reason for the table is indication of the state of the workspace: some actions are performed on a running workspace, while one of the actions (starting a stopped workspace) is performed on a stopped workspace, while the last action (deleting) is applicable to both running and stopped workspaces. I realize this info may appear trivial, but for example restarting a running workspace and starting a stopped workspace (which can also be defined as restarting a workspace) are two different actions that are taken using different UI elements.

.Basic actions you can perform on a workspace
|===
| Action | GUI steps in the Workspaces page

| _Reopen a running workspace_
| Click *Open*.

| _Restart a running workspace_
| Go to *⋮* *>* *Restart Workspace*.

| _Stop a running workspace_
| Go to *⋮* *>* *Stop Workspace*.

| _Start a stopped workspace_
| Click *Open*.

| _Delete a workspace_
| Go to *⋮* *>* *Delete Workspace*.

|===

> IMPORTANT: When you stop a workspace, its state at the moment you stopped it is preserved until you restart it. And if you do not stop a running workspace manually, it stops automatically when it reaches an idle timeout limit or a running timeout limit, which are set by the administrator of your organization's {prod-short} instance. When the workspace stops automatically because of a timeout limit, its latest state is preserved until you restart it.

Reviewer: ```suggestion
IMPORTANT: Stopping a workspace preserves its state until it restarts. This is true whether you stop the workspace or it reaches the idle timeout limit or running timeout limit set by your {prod-short} cluster administrator.
```

Reviewer: I'm not sure, but I think "the administrator of your organization's {prod-short} instance" and "cluster administrator" are equivalent.

Reviewer: You can remove most instances of "manually" and "automatically" from well-written technical documentation.

> IMPORTANT: When you stop a workspace, its state at the moment you stopped it is preserved until you restart it. And if you do not stop a running workspace manually, it stops automatically when it reaches an idle timeout limit or a running timeout limit, which are set by the administrator of your organization's {prod-short} instance. When the workspace stops automatically because of a timeout limit, its latest state is preserved until you restart it.

Reviewer 3: State is preserved only if it's not using ephemeral mode.

Using ephemeral storage, you loose any changes (state not preserved)

Author: Thank you for clarifying, Florent.
I'm removing this admonition because the section https://www.eclipse.org/che/docs/che-7/installation-guide/configuring-storage-types/ is not listed in https://github.com/eclipse/che/issues/21284.
